integrations/gitlab/gitlab_integration/gitlab_service.py

```diff
@@ -556,3 +568,79 @@ def get_entities_diff(
         logger.info(f"Found {len(entities_after)} entities in the current state")
 
         return entities_before, entities_after
+
+    def _parse_file_content(
+        self, file: ProjectFile
+    ) -> Union[str, dict[str, Any], list[Any]] | None:
+        """
+        Process a file from a project. If the file is a JSON or YAML, it will be parsed, otherwise the raw content will be returned
+        :param file: file object
+        :return: parsed content of the file
+        """
+        if file.size > MAX_ALLOWED_FILE_SIZE_IN_BYTES:
+            logger.warning(
+                f"File {file.file_path} is too large to be processed. Maximum size allowed is 1MB. Actual size of file: {file.size}"
+            )
+            return None
+        try:
+            return json.loads(file.decode())
+        except json.JSONDecodeError:
+            try:
+                documents = list(yaml.load_all(file.decode(), Loader=yaml.SafeLoader))
+                return documents if len(documents) > 1 else documents[0]
+            except yaml.YAMLError:
+                return file.decode().decode("utf-8")
+
+    async def get_and_parse_single_file(
+        self, project: Project, file_path: str, branch: str
+    ) -> dict[str, Any] | None:
+        try:
+            project_file = await AsyncFetcher.fetch_single(
+                project.files.get, file_path, branch
+            )
+            project_file = typing.cast(ProjectFile, project_file)
+            parsed_file = self._parse_file_content(project_file)
+            project_file_dict = project_file.asdict()
+
+            if parsed_file:
+                # Update the content with the parsed content. Useful for JSON and YAML files that can be further processed using itemsToParse
+                project_file_dict["content"] = parsed_file
+
+            return {"file": project_file_dict, "repo": project.asdict()}
+        except Exception as e:
+            logger.error(
+                f"Failed to process file {file_path} in project {project.path_with_namespace}. error={e}"
+            )
+            return None
+
+    async def get_all_files_in_project(
+        self, project: Project, path: str
+    ) -> typing.AsyncIterator[List[dict[str, Any]]]:
+        branch = project.default_branch
+        try:
+            file_paths = self._get_file_paths(project, path, branch)
+            logger.debug(
+                f"Found {len(file_paths)} files in project {project.path_with_namespace} files: {file_paths}"
```

Comment:
I know its not code you've written, but it isn't async while we can do it async. lets fix that